Automattic/pocket-casts-ios
# Sleep Timer Improvements: shake device to restart sleep timer

This PR adds the ability to shake the device so the sleep timer is restarted. This way, users can restart the sleep timer without unblocking their phones.

## Additional info

The traditional way of checking a device shake through `motionEnded` doesn't work if the app is in the background. The only way to achieve that is by using `CMMotionManager`.

I created an entity called `BackgroundShakeObserver` that only starts observing device motion while:

1. A timer is going on
2. The app is in the background

My tests show energy consumption shouldn't be higher, but we'll have to follow up.

## To test

As you can imagine, test on a real device. :)

Go to Profile > Settings > Beta features > enable tracksLogging.

1. Run the app
2. Start a timer
3. Shake the device
4. ✅ Timer should restart and you should see a toast saying "Sleep timer restarted due to device shake"
5. ✅ You should see `playerSleepTimerRestarted` being fired with `reason: device_shake`
7. Put the app on background and lock the phone
8. Shake the device
9. ✅ You should see `playerSleepTimerRestarted` being fired with `reason: device_shake`
10. ✅ A subtle confirmation sound should play
11. Open the app again and check the timer
12. ✅ It should have restarted

## Checklist
